draft-gruessing-moq-requirements.md

```diff
@@ -206,15 +207,25 @@ Our goal in this section is to understand the requirements that result from the
 
 ## Specific Protocol Considerations {#proto-cons}
 
-### Common push protocol for ingest and distribution
+In order to support the various topologies and patterns of media flows with the protocol, the protocol MUST support both sending and receiving of media streams, as separate actions or concurrently in a given connection.
 
-### Allow configurable latency - live streaming , interactive, media conferencing
+### Delivery Assurance vs. Delay
 
-### Support media transport mapping over QUIC Streams/QUIC Datagram (needs further evaluation)
+Different use cases have varying requirements with respect to the tradeoffs associated in having guarantee of delivery vs delay - in some (such as telefony) it may be acceptable to drop some or all of the media as a result of changes in network connectivity, throughput, or congestion whereas in other scenarios all media must arrive at the receiving end even if delayed. There SHOULD be support for some means for a connection to signal which media may be abandoned, and behaviours of both senders receivers defined when delay or loss occurs. Where multiple variants of media are sent, this SHOULD be done so in a way that provides pipelining so each media stream may be processed in parallel.
 
-### Support Webtransport/H3/Raw QUIC as media transport
+### Support Webtransport/Raw QUIC as media transport
 
-### Supports media over QUIC Streams/QUIC Datagrams
+Many of the use cases will be deployed in contexts where web browsers are the common application runtime; thus the use of existing protocols and APIs is desireable for implementations. Support for WebTransport {{I-D.draft-ietf-webtrans-overview}} SHOULD be defined, however SHOULD NOT be required by all implementations or deployments.
+
+Considerations should be made clear with respect to modes where WebTransport "falls back" to using HTTP/2 or other future non-QUIC based protocol.
+
+### Media Negotiation & Agility {#MOQ-negotiation}
+
+All entities which directly process media will have support for a variety of media codecs, conversely there SHOULD be capability in the protocol for sender and receiver to negotiate which media codecs will be used in a given session, as well as defining behaviours when one or more is not supported. Media encryption {{MOQ-media-encryption}} should also be factored in as part of the negotiation. Renegotiation SHOULD also be possible, allowing for changes in codec within an existing session. Consideration should be made if relays are also able to facilitate negotation even if not directly processing media.
```

Comment:
> Consideration should be made if relays are also able to facilitate negotation even if not directly processing media.

Keeping in mind that we really need an agreed basic architecture badly, is this better stated as 

> Consideration should be made if caching relays are also able to facilitate negotiation if not directly processing media.

Nit: that should be "negotiation".

Reply:
Why specifically caching relays? I think media negotiation is somewhat orthogonal to any cacheing capability.

Reply:
Perhaps I lack imagination. I was mostly trying to think about why non-caching relays would need to negotiate a different codec, or even a different variant of the media. We are pushing on how much a relay can do without being able to decode media metadata, but we haven't worked that out yet. If this turns out to be part of rate adaptation, for instance, and relays are supposed to do that, then it would be incorrect to say "caching relays" here. 

It's fine with me to drop that for now.